packages/lib/src/toolbar/controls/DomainWidget/DomainWidget.tsx

```diff
@@ -106,48 +106,62 @@ function DomainWidget(props: Props) {
           onToggle={() => toggleEditing(!isEditing)}
         />
       </div>
-      <DomainTooltip
-        ref={tooltipRef}
+
+      <div
         id={TOOLTIP_ID}
-        open={hovered || isEditing}
-        sliderDomain={sliderDomain}
-        dataDomain={dataDomain}
-        errors={errors}
-        isAutoMin={isAutoMin}
-        isAutoMax={isAutoMax}
-        onAutoMinToggle={() => {
-          const newMin = isAutoMin ? dataDomain[0] : null;
-          onCustomDomainChange([newMin, customDomain[1]]);
-          if (!isAutoMin) {
-            toggleEditingMin(false);
-          }
-        }}
-        onAutoMaxToggle={() => {
-          const newMax = isAutoMax ? dataDomain[1] : null;
-          onCustomDomainChange([customDomain[0], newMax]);
-          if (!isAutoMax) {
-            toggleEditingMax(false);
-          }
-        }}
-        isEditingMin={isEditingMin}
-        isEditingMax={isEditingMax}
-        onEditMin={toggleEditingMin}
-        onEditMax={toggleEditingMax}
-        onChangeMin={(val) => onCustomDomainChange([val, customDomain[1]])}
-        onChangeMax={(val) => onCustomDomainChange([customDomain[0], val])}
-        onSwap={() => onCustomDomainChange([customDomain[1], customDomain[0]])}
+        className={styles.tooltip}
+        role="dialog"
+        aria-label="Edit domain"
+        hidden={!hovered && !isEditing}
       >
-        {histogram && (
-          <Histogram
+        <div className={styles.tooltipInner}>
+          {histogram && (
+            <Histogram
```

Comment:
The diff is crap, but the move of the tooltip logic is fairly straightforward. The only thing worth mentioning is that the `Histogram` is no longer passed as `children` prop of `DomainTooltip` (now `DomainControls`).

It's `DomainWidget` that takes care of rendering `Histogram` directly in the body of the tooltip (`.tooltipInner`) and of laying it and `DomainControls` out appropriately with CSS.